pkg/datapath/linux/probes/probes_privileged_test.go

```diff
@@ -48,3 +52,26 @@ func (s *ProbesPrivTestSuite) TestHelpers(c *C) {
 	_, ok := pm.GetHelpers("sched_act")["bpf_map_lookup_elem"]
 	c.Assert(ok, Equals, true)
 }
+
+func (s *ProbesPrivTestSuite) TestWriteHeaders(c *C) {
+	var buf bytes.Buffer
+
+	expectedPatterns := []string{
+		"#ifndef BPF_FEATURES_H_",
+		"#define BPF_FEATURES_H_",
+		"#define HAVE_BPF_SYSCALL",
+		"#define HAVE.*PROG_TYPE",
+		"#define HAVE.*MAP_TYPE",
+		"#define HAVE.*HELPER",
+		"#endif /\\* BPF_FEATURES_H_ \\*/",
```

Comment:
Maybe add another regular expression here to match the `#define BPF__PROG_TYPE_.*__HELPER_.* [01]` cases?